commands/01-mod.js

```diff
@@ -158,18 +158,37 @@ const Module = new Augur.Module()
           if (bannerHighRole.comparePositionTo(bannedHighRole) <= 0) {
             msg.reply(`you cannot ban ${u.escapeText(member.displayName)}!`);
             continue;
-          } else {
+          }
+          members.set(userId, member);
+        }
+        else {
+          members.set(userId, undefined);
+        }
+      }
+    }
+    if (members.size == 0) {
+      msg.reply("you need to tell me who to ban!").then(u.clean);
+      return;
+    }
+
+    let confirm = await u.confirm(msg, `Are you sure you want to ban the following?\n${members.keyArray().map(m => members.get(m) ? u.escapeText(members.get(m).displayName) : m).join("\n")}`);
+
+    if (confirm) {
+      let banCount = 0;
+      for (const [memberId, member] of members) {
+        try {
+          if (member) {
             const infraction = {
               discordId: member.id,
               description: reason,
               value: 30,
               mod: msg.author.id
             };
-            await (member.send(`You were banned from ${msg.guild.name} for ${reason}`).catch(() => blocked(member)));
+            let inf = await Module.db.infraction.save(infraction);
 
+            await (member.send(`You were banned from ${msg.guild.name} for ${reason}`).catch(() => blocked(member)));
             if (!msg.client.ignoreNotifications) msg.client.ignoreNotifications = new Set();
             msg.client.ignoreNotifications.add(member.id);
-
             await member.ban({days: 2, reason});
 
             let embed = u.embed()
```

Comment:
Get a user we can use for the embeds, whether member present or not. For reasons unknown, github isn't letting me comment on the next three lines of the embed. We don't want duplicated content.
```suggestion
            let user = member?.user || await msg.client.users.fetch(memberId);
            let embed = u.embed()
              .setAuthor(user.username, user.displayAvatarURL({dynamic: true}))
              .setTitle("User Ban")
              .setDescription(`**${u.escapeText(msg.member.displayName)}** banned **${u.escapeText(user.username)}** for ${reason}.`)
```

Reply:
Do we want a catch on that `msg.client.users.fetch(memberId);`?

Reply:
yeah, wouldn't hurt to have a `.catch(u.noop)` on there.